src/main/java/com/inin/analytics/elasticsearch/ESEmbededContainer.java

```diff
@@ -127,65 +135,60 @@ public void deleteSnapshot(String snapshotName, String snapshotRepoName) {
-			.put("indices.fielddata.cache.size", "0%");
+            .put("path.repo", snapshotWorkingLocation)
+            .put("path.home", workingDir)
+            .put("bootstrap.memory_lock", true)
+			.put("cluster.routing.allocation.disk.watermark.low", "99%") // Nodes don't form a cluster, so routing allocations don't matter
+			.put("cluster.routing.allocation.disk.watermark.high", "99%")
+            .put("node.data", true)       //data node
+            .put("transport.type", "local")
+            .put("indices.memory.index_buffer_size", "5%") // The default 10% is a bit large b/c it's calculated against JVM heap size & not Yarn container allocation. Choosing a good value here could be made smarter.
+            .put("indices.fielddata.cache.size", "0%")
+            .put("cluster.name", clusterName);
 			
-			if(memoryBackedIndex) {
-				builder.put("index.store.type", "memory");
-			}
 			Settings nodeSettings = builder.build();
 
-			// Create the node
-			container.setNode(nodeBuilder()
-					.client(false) // It's a client + data node
-					.local(true) // Tell ES cluster discovery to be inner-jvm only, disable HTTP based node discovery
-					.clusterName(clusterName)
-					.settings(nodeSettings)
-					.build());
+			try {
+                //read plugin list
+			    List<Class<? extends Plugin>> pluginClasses = container.getPluginClasses(customPluginListFile);
 
-			// Start ES
-			container.getNode().start();
+                // Create the node
+                container.setNode(new PluginConfigurableNode(nodeSettings, pluginClasses));
 
-			// Configure the cluster with an index template mapping
-			if(templateName != null && templateSource != null) {
-				container.getNode().client().admin().indices().preparePutTemplate(templateName).setSource(templateSource).get();	
-			}
+	            // Start ES
+                container.getNode().start();
 
-			// Create the snapshot repo
-			if(snapshotWorkingLocation != null && snapshotRepoName != null) {
-				Map<String, Object> settings = new HashMap<>();
-				settings.put("location", snapshotWorkingLocation);
-				settings.put("compress", true);
-				settings.put("max_snapshot_bytes_per_sec", "400mb"); // The default 20mb/sec is very slow for a local disk to disk snapshot
-				container.getNode().client().admin().cluster().preparePutRepository(snapshotRepoName).setType("fs").setSettings(settings).get();
-			}
+                // Set default index setting
+                container.setDefaultIndexSettings();
+                
+                // Configure the cluster with an index template mapping
+                if(templateName != null && templateSource != null) {
+    		         container.getNode().client().admin().indices().preparePutTemplate(templateName).setSource(templateSource).get();
+    			}
+
+    			// Create the snapshot repo
+    			if(snapshotWorkingLocation != null && snapshotRepoName != null) {
+    				Map<String, Object> settings = new HashMap<>();
```

Comment:
Do we need it here? Not sure how much benefit we can get. It seems default value would be fine.